### PR TITLE
Convert Person from id based on personnr to member_no

### PR DESCRIPTION
**Pull request by martin-green**
Implemented add person to scoutnet queue to use the returned member_no.
Implemented a migration script
You can not add a "local person", using personnr. You have to use add to scoutnet queue to get a member id.
Added information on missing add to scoutnet queue api-key.
Lots of optimizations to make the conversion to run quickly.
* Progress report nolonger stored in datastore. Using memcache to get better performance.
* Progress reporting was O(N^2) for number of report items. A cursor is used to only update new lines.
Moved admin utilities to new admin.py file
Moved progress related utilities to progress.py
Renamed importresult.html to progressreport.html
Add a link to scout group after import is done

### CODE REVIEW

File: scoutnet.py [hunk 1]
```diff
@@ -152,15 +195,29 @@ def AddPersonToWaitinglist(scoutgroup, firstname, lastname, personnummer, emaila
...
-        #j = json.loads(result_json)
         raise ScoutnetException(result_json.decode('unicode_escape')) # display the raw json message
-        return False
+        return 0
 
     if 200 <= response.getcode() <= 201:
         result_json = response.read()
         logging.info("Added person: " + result_json)
+        # response = """{"ss":{"member_no":111111,"sex":"1|2","date_of_birth":"YYYY-MM-DD","first_name":"xx","last_name":"yy","email":"xx@yy.se","newsletter":"1","preferred_culture":"sv","ssno":"nnnn","nick_name":null,"note":null,"product_subscription_8":"1","id":null},"membership":{"troop_id":null,"status":2,"feegroup_id":null,"patrol_id":null},"contact_list":{"contacts":{"contact_9":{"details":"0733-411 815","contact_type_id":"39"},"contact_1":{"contact_type_id":"1","details":"0763-967 792"},"contact_3":{"details":"Sophie Engstr\u00f6m","contact_type_id":"14"},"contact_5":{"details":"0763-967 792","contact_type_id":"38"}}},"address_list":{"addresses":{"address_1":{"country_code":"752","zip_name":"City","address_type":"0","zip_code":"#####","is_primary":"1","address_line1":"gatan 1"}}}}"""
```
Comment (martin-green) [inline scoutnet.py:204]: Den här informationen måste tas bort!

**Review by tobbee — COMMENTED** [review]
Impressive. I thought it would be much more straightforward to make the change, but part of the massive PR is that you've moved things around and cleaned some stuff.

I've read all changes through pretty quickly, and it looks good, but hard to know if any details are off.

 I'll try to check it once more tomorrow.

**Comment by TobbeEdgeware** [comment]
Jag har provkört på en lokal dev-server och där fungerade det bra att byte från personnummer till medlemsnummer. 👍

File: templates/admin.html [hunk 2]
```diff
@@ -6,15 +6,23 @@
...
 			<div class="btn-toolbar">
 				<a href="/admin/backup/" class="btn btn-lg btn-default">Backup xml</a>
 			</div>
+			<hr/>
+			<div class="btn-toolbar">
+				<a href="/admin/merge_sg/" class="btn btn-lg btn-default">Byt kårnamn</a>
+			</div>
+			<div class="btn-toolbar">
+				<a href="/admin/update_person_ids/" class="btn btn-lg btn-default">Byt från personnr id på all Personer</a>
+			</div>
```
Comment (TobbeEdgeware) [inline templates/admin.html:22]: Skriv om till "Byt person id från personnummer till medlemsnummer"

File: templates/update_person_ids.html [hunk 3]
```diff
@@ -0,0 +1,28 @@
+﻿{% include "header.html" %}
+<body>
+{% include "navigation.html" %}
+	<div class="main">
+		<div class="container">
+		<form id="update_person_ids" class="form-horizontal" role="form" name="person" method="POST">
+		  <h1>Updatera Person IDn</h1>
+			<p>Byter ut personer som har personnr som id till medlemsnummer.</p>
+			<div class="form-group row">
```
Comment (TobbeEdgeware) [inline templates/update_person_ids.html:9]: Skriv om till "Byt personer med id=personnummer till id=medlemsnr"

**Review by TobbeEdgeware — APPROVED** [review]
Det ser bra ut. Har en liten anmärkning om texten som beskriver vad som görs.
🥇

**Review by TobbeEdgeware — APPROVED** [review]
Råkade godkänna med jobb-githubkonto, så godkänner här med privata kontot också 🥇

**Pull request merged**